agent/config/config.go

```diff
@@ -181,6 +183,39 @@ func FileConfig() Config {
 	return config
 }
 
+func parseEnvVariableInt(envVar string, defaultValue uint16) (uint16) {
```

Comment:
Nitpick: Can you move `parseEnvVariableInt` and `parseEnvVariableDuration` below `EnvironmentConfig`?  One of the things I try to do in terms of code organization is have the caller above the callee so that a reader sees the highest abstraction layer first and can go further down for details when necessary.